include/rdma/fabric.h

```diff
@@ -315,6 +315,35 @@ enum {
 	FI_PROTO_EFA
 };
 
+#define FI_TC_TYPE_RESERVED_BITS 	4
+#define FI_TC_TYPE_MASK 		(0xf)
+#define FI_TC_VALUE_RESERVED_BITS 	28
+#define FI_TC_VALUE_MASK 		(0x0fffffff)
+#define FI_TC_PROV_SPECIFIC_BIT 	(0x08000000)
+#define FI_TC_LABEL_MASK 		(0x07ffffff)
```

Comment:
Some early feedback on your update:
FI_PROV_SPECIFIC is already a defined value in the header file.  We don't need to treat the tclass field as bit fields.  It's a 32-bit value separated into ranges.  Range 0-63 map to DSCP.  Another range maps to what you're calling 'label' values.  The label range can start anywhere, but if we want to remove the gaps between DSCP and label values, the labels can start at 64, though it may make sense to start at 256.  We don't really need anything more complex than:
```
enum {
    /* Values 0-255 map to DSCP */
    FI_TC_BEST_EFFORT = 255,
    FI_TC_LOW_LATENCY,
    ....
};
```
There's really no advantage to having type + value fields.

Reply:
Sounds fine to me. If we want to make the field offset-based, rather than with reserved bits, how much space would you be comfortable allocating to libfabric labels? I was thinking somewhere around 256 for libfabric-defined, and 256 for provider defined. 

I can't imagine a scenario where we'd ever need that many, but we aren't running low on space either.

Reply:
At this point, we don't need to define a range limit for labels.  They will only consume the range of values in the enum, which will be something like 256 to 260-some.  (Btw, my 255 above should start at 256).  Anything above the last enum value is reserved for future use, with the exception if the upper bit (FI_PROV_SPECIFIC) is set.  Basically, the decision is deferred until such time that we actually need it, if ever, at which time we'll have more information on the requirements of the new range.

Reply:
Sounds good to me.